test/testdata/resolver/type_alias_not_on_T.rb

```diff
@@ -0,0 +1,15 @@
+# typed: true
+
+module A
+  extend T::Sig
+
+  module NotT
+    def self.type_alias(&blk); end
+  end
+
+  NotATypeAlias = NotT.type_alias {Object}
+
+  sig {returns(NotATypeAlias)}
+  #            ^^^^^^^^^^^^^ error: Constant `A::NotATypeAlias` is not a class or type alias
```

Comment:
Side note: This is kind of a strange error message.

The message suggests it should be a "class or type alias". But a `module` would also be valid...

```suggestion
  #            ^^^^^^^^^^^^^ error: Constant `A::NotATypeAlias` is not a class, module or type alias
```

It could be a few other things like a type union, type intersection, ..., so perhaps it would be best to just:

```suggestion
  #            ^^^^^^^^^^^^^ error: Constant `A::NotATypeAlias` is not a type
```

Reply:
Good point. It's subtle because the "class" in "class or type alias" does not refer to `class` the keyword but rather to "class aliases" which are what Sorbet calls constants defined like `ClassAlias = Integer` (as opposed to other constants, like `X = 1`). Class aliases can alias to modules as well.

Ruby natively doesn't have a great name for "constant which is an alias to a class, module or other such constant alias." The name it uses exclusively internally and in the docs for these constants is "class alias" (see e.g. https://srb.help/5031).

Sorbet already knows that this constant is _not_ a reference to other kinds of constants which would be valid in type syntax. The full list of constant types allowed in type syntax are:

- Names of classes and modules (no aliases)
- Class alias constants
- Type alias constants
- Generic type members
- Enum values

The types of constants which are not allowed:

- Static field constants ("normal" constant assignments)

The message does further clarify how to fix this error:

```
If you are trying to define a type alias, you should use T.type_alias here
```

and points to the definition of the static field constant.

I could see "is not a type" being a better framing of this error than "is not a class or type alias," though I would love if there were an error message that did a better job of educating users about the subtleties of different kinds of constants allowed in type syntax.

Reply:
> The name it uses exclusively internally and in the docs for these constants is "class alias" (see e.g. https://srb.help/5031).

Should this error message match that, then?

```suggestion
  #            ^^^^^^^^^^^^^ error: Constant `A::NotATypeAlias` is not a class alias or type alias
```

Reply:
That would be another option yes